### PR TITLE
feat: invoke `flush_table` and `compact_table` in fuzz tests

### LINKED ISSUE #4046: Failed to flush physical metric table
### What type of bug is this?

Unexpected error

### What subsystems are affected?

Table Engine

### Minimal reproduce step

1. Switch to https://github.com/GreptimeTeam/greptimedb/pull/4045
2. Run GreptimeDB
3. Run 
```
cargo fuzz run --fuzz-dir tests-fuzz -D --sanitizer=none fuzz_insert_logical_table
```

### What did you expect to see?

Flush data successfully

### What did you see instead?

2024-05-26T10:56:44.595514Z  INFO operator::request: Handle table manual compaction request: CompactTableRequest { catalog_name: "greptime", schema_name: "public", table_name: "et", compact_options: Regular(Regular) }
2024-05-26T10:56:44.595696Z ERROR servers::mysql::writer: Failed to handle mysql query err=0: , at src/common/recordbatch/src/adapter.rs:258:55
1: External(0: Failed to do table mutation, at src/operator/src/table.rs:97:14
1: Failed to send request to region, at src/operator/src/request.rs:182:22
2: External error, at src/frontend/src/instance/standalone.rs:86:14
3: Operation to region server failed, at src/frontend/src/instance/standalone.rs:68:14
4: Execute gRPC request error, at src/datanode/src/region_server.rs:258:22
5: Failed to handle request for region 4398046511104(1024, 0), at src/datanode/src/region_server.rs:499:14
6: Unsupported region request: Compact, at src/metric-engine/src/engine.rs:149:87)

### What operating system did you use?

Doesn't matter

### What version of GreptimeDB did you use?

main

### Relevant log output and stack trace

_No response_

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)

close #4046

## What's changed and what's your intention?

Invoke `flush_table` and `compact_table` in fuzz tests

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: tests-fuzz/targets/fuzz_insert_logical_table.rs [hunk 1]
```diff
@@ -225,26 +220,62 @@ async fn execute_insert(ctx: FuzzContext, input: FuzzInput) -> Result<()> {
...
+        let logical_table_ctx = Arc::new(TableContext::from(&create_logical_table_expr));
+
+        insert_and_validate_values(&ctx, &mut rng, logical_table_ctx.clone()).await?;
+        tables.insert(logical_table_ctx.name.clone(), logical_table_ctx);
+        if rng.gen_bool(0.1) {
+            flush_memtable(&ctx.greptime, &physical_table_ctx.name).await?;
+        }
+        if rng.gen_bool(0.1) {
+            compact_table(&ctx.greptime, &physical_table_ctx.name).await?;
+        }
```
Comment (killme2008) [inline tests-fuzz/targets/fuzz_insert_logical_table.rs:257]: I think we should validate the rows in tables after flushing or compacting tables.

**Review by killme2008 — COMMENTED** [review]
Good job!

**Review by fengjiachun — APPROVED** [review]
LGTM

**Review by evenyag — APPROVED** [review]
LGTM

**Pull request merged**